### DISCUSSION TITLE
Renaming image files based on iNat Metadata

### DISCUSSION THREAD

**Discussion by jjmontem** [rcloran/lr-inaturalist-publish#6] [Ideas]
Is it possible to use the metadata written from iNaturalist using this plugin to rename files? If I'd like to batch rename my images, I see some metadata fields can be used, but this excludes the plugin-specific fields.

**Comment by rcloran** [comment]
Unfortunately I don't think there's anything that I could do to allow use of the plugin-specific fields. But, I don't think it would be unreasonable to have an option to set the title to the identified species, which you could then use to do the rename. Would that work for you?

**Reply by rcloran** [reply]
Some further comments on a similar topic from someone who has developed quite a few Lightroom plugins, and so has more experience with it than I do: http://regex.info/blog/lightroom-goodies#comment-60782

**Reply by rcloran** [reply]
Here's another option, using a separate paid plugin: http://lightroomsolutions.com/articles/complex-file-renaming-in-lightroom-made-simpler/

It's effectively the same idea as my suggestion that lr-inaturalist-publish could set the title, but a bit more flexible, and a bit more costly.

**Reply by rcloran** [reply]
Another plugin that could be used to copy from some fields to others. I wasn't able to find a way to copy plugin metadata, so it might just give a similar problem: http://regex.info/blog/lightroom-goodies/bag-o-goodies#write-field

**Comment by jjmontem** [comment]
I’ll have to verify that the Title field is an available option in Lightroom’s batch renaming feature, but if it is, yes. I’d like to name photographs based on their lowest taxonomic identifier.

I’ll check out those other plugins too…. Might be a while as the academic year is wrapping up…


--

Justin Montemarano, PhD (he/him)

Department of Biological Sciences

Colorado Mesa University

Wubben Science 211F

https://www.researchgate.net/profile/Justin_Montemarano



Chat with me on ***@***.***>!
Make an appointment<https://calendly.com/jmontemarano/studentoffice>!

________________________________

**Reply by rcloran** [reply]
> I’ll have to verify that the Title field is an available option in Lightroom’s batch renaming feature, but if it is, yes. 

Yeah, title is available. As is "Headline", which the author of the first plugin I mentioned suggests due to its low use. 

> I’d like to name photographs based on their lowest taxonomic identifier.

I think that trying to do this by keyword might be difficult, so I'll take a look at an option to set the title (or headline) during sync.

**Comment by rcloran** [comment]
[0.4.1-dev.3](https://github.com/rcloran/lr-inaturalist-publish/releases/download/nightly/lr-inaturalist-publish-0.4.1-dev.3.g355032c.zip) and newer have an option to set the title on sync. A full sync should set all titles.

**Comment by dominictinner** [comment]
This feature somehow doesn't work for me. I get the following error when trying to sync in the plugin settings: [string "SyncObservations.lua"]:158: bad argument #1 to 'pairs' (table expected, got nil).

**Reply by rcloran** [reply]
;( Could you create an issue for that and include the debug logs for the sync? As far as I can tell the issue isn't really related to the title, but to the keyword sync generally -- and I'd bet it's triggered by https://www.inaturalist.org/observations/164270315 since that has no ID, and I bet I messed up by not considering that case properly.

What would be most helpful to me is to see the logs from a sync as things stand, and then have you add an ID to that observation (even if it's just "Birds"), and try again, and let me know if that fixes things.

**Reply by rcloran** [reply]
The problem where observations with no ID caused a sync failure should now be fixed in v0.5.1.

Commit: https://github.com/rcloran/lr-inaturalist-publish/commit/0f449d3c121cf7c66d898f4901234429696ed4b5
